feature-libs/order/core/facade/reorder-order.service.ts

```diff
@@ -48,4 +52,24 @@ export class ReorderOrderService implements ReorderOrderFacade {
       orderId,
     });
   }
+
+  protected reorderPreconditions(): Observable<string> {
+    return combineLatest([
+      this.userIdService.takeUserId(),
+      this.activeCartFacade.getActiveCartId(),
+    ]).pipe(
+      take(1),
+      map(([userId, cartId]) => {
+        if (!userId) {
+          throw new Error('Must be logged in to reorder');
+        }
+
+        if (cartId) {
+          this.multiCartFacade.deleteCart(cartId, userId);
```

Comment:
Before reorder, the active cart will be removed? Can we borrow the idea of restore saved cart? (The saved cart becomes active cart, while the current active cart will be saved)

Reply:
and if reorder is successful, then save the active cart.

Reply:
maybe confirm with Bill, how to handle the active cart for reorder.

Reply:
As I mentioned in the DM, but i'll paste here for visibility.

1. Bill said to delete as this was the first phase of the feature
2. future improvement is to optionally allow users to the save the cart

All in all, we'll delete it for now, and future improvements may be done for this feature.